presto-elasticsearch/src/main/java/com/facebook/presto/elasticsearch/ElasticsearchColumnHandle.java

```diff
@@ -0,0 +1,145 @@
+import static java.util.Objects.requireNonNull;
+
+public final class ElasticsearchColumnHandle
+        implements ColumnHandle
+{
+    private final String columnName;
+    private final Type columnType;
+    private final String columnJsonPath;
+    private final String columnJsonType;
+    private final int ordinalPosition;
+    private final boolean isList;
+
+    @JsonCreator
+    public ElasticsearchColumnHandle(
+            @JsonProperty("columnName") String columnName,
+            @JsonProperty("columnType") Type columnType,
+            @JsonProperty("columnJsonPath") String columnJsonPath,
+            @JsonProperty("columnJsonType") String columnJsonType,
+            @JsonProperty("ordinalPosition") int ordinalPosition,
+            @JsonProperty("isList") boolean isList)
+    {
+        this.columnName = requireNonNull(columnName, "columnName is null");
+        this.columnType = requireNonNull(columnType, "columnType is null");
+        this.columnJsonPath = requireNonNull(columnJsonPath, "columnJsonPath is null");
+        this.columnJsonType = requireNonNull(columnJsonType, "columnJsonType is null");
+        this.ordinalPosition = ordinalPosition;
+        this.isList = isList;
+    }
+
+    @JsonProperty
+    public String getColumnName()
+    {
+        return columnName;
+    }
+
+    @JsonProperty
+    public Type getColumnType()
+    {
+        return columnType;
+    }
+
+    @JsonProperty
+    public String getColumnJsonPath()
+    {
+        return columnJsonPath;
+    }
+
+    @JsonProperty
+    public String getColumnJsonType()
+    {
+        return columnJsonType;
+    }
+
+    @JsonProperty
+    public int getOrdinalPosition()
+    {
+        return ordinalPosition;
+    }
+
+    @JsonProperty
+    public boolean getIsList()
+    {
+        return isList;
+    }
+
+    public ColumnMetadata getColumnMetadata()
+    {
+        Map<String, Object> properties = new HashMap<>();
+        properties.put("jsonPath", columnJsonPath);
+        properties.put("jsonType", columnJsonType);
+        properties.put("isList", isList);
+        properties.put("ordinalPosition", ordinalPosition);
+        return new ColumnMetadata(columnName, columnType, "", "", false, properties);
+    }
+
+    @Override
+    public int hashCode()
+    {
+        return Objects.hash(
+                columnName,
+                columnType,
+                columnJsonPath,
+                columnJsonType,
+                ordinalPosition,
+                isList);
+    }
+
+    @Override
+    public boolean equals(Object obj)
+    {
+        if (this == obj) {
+            return true;
+        }
+        if ((obj == null) || (getClass() != obj.getClass())) {
+            return false;
+        }
+
+        ElasticsearchColumnHandle other = (ElasticsearchColumnHandle) obj;
+        return Objects.equals(this.getColumnName(), other.getColumnName()) &&
+                Objects.equals(this.getOrdinalPosition(), other.getOrdinalPosition()) &&
```

Comment:
nit: move ordinal position before `isList`, which is the case for other usages in this class.

Reply:
sure, will do